# RuntimeException：运行时异常

# Exception：编译异常

## 自定义运行时异常

```java
// 类继承RuntimeException 为 运行时异常
// 提供两个构造方法，一个无参，一个有参
class MyException extends RuntimeException 为{
    public MyException() {
    }
    //String字符串参数是异常的信息
    public MyException(String message) {
        super(message);
    }
}
```

## 自定义编译异常

```java
// 类继承Exception 为编译异常
public class IllegalGenderException extends Exception{
    public IllegalGenderException() {
    }

    public IllegalGenderException(String message) {
        super(message);
    }
}
```

## 使用自定义异常

```java
public class Test {
    public static void main(String[] args) {
    	People people = new People();
        try {
            people.setAge(160);
        } catch (MyException my){
             my.printStackTrace();
        }
    
    }
    static class People{
        public void setAge(Integer age) {
            if (age > 150) {
                throw new MyException(age+"不合法，太大了");
            }
        }
    }
}
```

